openapi-contracts/src/main/com/sinch/sdk/domains/conversation/adapters/api/v1/AppApi.java

```diff
@@ -0,0 +1,466 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+
+public class AppApi {
+
+  private static final Logger LOGGER = Logger.getLogger(AppApi.class.getName());
+  private HttpClient httpClient;
+  private ServerConfiguration serverConfiguration;
+  private Map<String, AuthManager> authManagersByOasSecuritySchemes;
+  private HttpMapper mapper;
+
+  public AppApi(
+      HttpClient httpClient,
+      ServerConfiguration serverConfiguration,
+      Map<String, AuthManager> authManagersByOasSecuritySchemes,
+      HttpMapper mapper) {
+    this.httpClient = httpClient;
+    this.serverConfiguration = serverConfiguration;
+    this.authManagersByOasSecuritySchemes = authManagersByOasSecuritySchemes;
+    this.mapper = mapper;
+  }
+
+  /**
+   * Create an app You can create a new Conversation API app using the API. You can create an app
+   * for one or more channels at once. The ID of the app is generated at creation and will be
+   * returned in the response.
+   *
+   * @param projectId The unique ID of the project. You can find this on the [Sinch
+   *     Dashboard](https://dashboard.sinch.com/convapi/apps). (required)
+   * @param appCreateRequestDto The app to create. (required)
+   * @return AppResponseDto
+   * @throws ApiException if fails to make API call
+   */
+  public AppResponseDto appCreateApp(String projectId, AppCreateRequestDto appCreateRequestDto)
+      throws ApiException {
+
+    LOGGER.finest(
+        "[appCreateApp]"
+            + " "
+            + "projectId: "
+            + projectId
+            + ", "
+            + "appCreateRequestDto: "
+            + appCreateRequestDto);
+
+    HttpRequest httpRequest = appCreateAppRequestBuilder(projectId, appCreateRequestDto);
+    HttpResponse response =
+        httpClient.invokeAPI(
+            this.serverConfiguration, this.authManagersByOasSecuritySchemes, httpRequest);
+
+    if (HttpStatus.isSuccessfulStatus(response.getCode())) {
+      TypeReference<AppResponseDto> localVarReturnType = new TypeReference<AppResponseDto>() {};
+      return mapper.deserialize(response, localVarReturnType);
+    }
+    // fallback to default errors handling:
+    // all error cases definition are not required from specs: will try some "hardcoded" content
+    // parsing
+    throw ApiExceptionBuilder.build(
+        response.getMessage(),
+        response.getCode(),
+        mapper.deserialize(response, new TypeReference<HashMap<String, ?>>() {}));
+  }
+
+  private HttpRequest appCreateAppRequestBuilder(
+      String projectId, AppCreateRequestDto appCreateRequestDto) throws ApiException {
+    // verify the required parameter 'projectId' is set
+    if (projectId == null) {
+      throw new ApiException(
+          400, "Missing the required parameter 'projectId' when calling appCreateApp");
```

Comment:
I'm wondering whether is make sense to reference an internal method name in the exception message that will bubble up to the user

Reply:
Right. Thank you
This is coming from DTO generator tool: created a ticket for it